src/EVM/Types.hs

```diff
@@ -1605,6 +1605,19 @@ untilFixpoint f a =
 bsToHex :: ByteString -> String
 bsToHex bs = concatMap (paddedShowHex 2) (BS.unpack bs)
 
+-- Used during forceAddr to deal with symbolic addresses
+forceEAddrToEWord :: Expr EAddr -> Expr EWord
+forceEAddrToEWord = \case
+  LitAddr a -> Lit (into a)
+  SymAddr a ->  WAddr (SymAddr a)
+  _ -> internalError "Unexpected forceAddr"
+
+forceEWordToEAddr :: Expr 'EWord -> Expr 'EAddr
+forceEWordToEAddr = \case
+  Lit a -> LitAddr (truncateToAddr a)
+  WAddr (SymAddr a) -> SymAddr a
+  _ -> internalError "delegateCall: expected concrete address"
```

Comment:
Should there be a different error message?
Now that this is a standalone function, are we sure this will always be called in the context of `delegateCall`?

Reply:
Thanks, yeah, I have now updated the error messages for both `forceEWordToEAddr` and `orceEAddrToEWord`.